core/templates/pages/exploration-editor-page/history-tab/history-tab.component.ts

```diff
@@ -180,11 +180,23 @@ angular.module('oppia').component('historyTab', {
           ctrl.selectedVersionsArray[0], ctrl.selectedVersionsArray[1]);
         var laterComparedVersion = Math.max(
           ctrl.selectedVersionsArray[0], ctrl.selectedVersionsArray[1]);
-
-        ctrl.compareVersionMetadata.earlierVersion =
-          ctrl.totalExplorationVersionMetadata[earlierComparedVersion - 1];
-        ctrl.compareVersionMetadata.laterVersion =
-          ctrl.totalExplorationVersionMetadata[laterComparedVersion - 1];
+        let earlierIndex = null, laterIndex = null;
+        for (let i = 0; i < ctrl.totalExplorationVersionMetadata.length; i++) {
```

Comment:
Please add more frontend tests for this logic. I feel like the issue happened originally because the coverage wasn't as good. Ideally I would expect more changes in the spec file for this change in logic.

Reply:
Yeah right. I have added anothe ```it``` block to test this logic.